tools/redis-tools.sh

```diff
@@ -0,0 +1,124 @@
+#   ./redis-tools.sh keys [Field Name] [Field Value]
+#       The matching keys are printed out.
+#   ./redis-tools.sh values [Field Name] [Field Value]
+#       The values of the matching keys are printed out.
+#   ./redis-tools.sh delete [Field Name] [Field Value]
+#       The matching keys are deleted from redis.
+#   ./redis-tools.sh dump [Field Name] [Field Value] [Directory to Dump]
+#       The matching keys are dumped to the indicated directory.
+#   ./redis-tools.sh restore [Directory to Restore] 
+#       The keys dumped in the directory are restored.
+#
+# Examples: 
+#  ./redis-tools.sh keys all
+#  ./redis-tools.sh values id f5bd194d9424427c9d82a57eedc531d0
+#  ./redis-tools.sh dump request_id 09879dbc90a44c6db80dc9f68113d829 storagefolder
+#  ./redis-tools.sh delete all
+#  ./redis-tools.sh restore storagefolder
+#  ./redis-tools.sh dump all storagefolder
+
+action=$1 # "keys" / "values" / "delete" / "dump" / "restore"
+
+case "$action" in
+  "restore")
+    directory=$2
+    if [[ -z "$directory" ]]; then
+      >&2 echo Error: directory not provided
+    else
+      for filepath in "$directory"/*; do
+        key=$(basename "$filepath")
+        echo Restoring "$key"
+        cat "$filepath" | redis-cli -x restore "$key" 0
+      done
+    fi
+  ;;
+
+  "keys"|"values"|"dump"|"delete")
+    # Ignores [Field Value] if "all" is passed
+    field_name="$2"
+    if [[ "$field_name" == "all" ]]; then
+      directory="$3";
+    else
+      field_value="$3"
+      directory="$4"
+    fi
+
+    if [[ "$action" == "dump" ]] && [[ -z "$directory" ]]; then
+      >&2 echo Error: directory not provided
+    else
+      if [[ "$action" != "values" ]]; then
+        declare -a key_array
+        if [[ "$action" == keys ]] || [[ "$field_name" != "all" ]]; then
+          echo Keys found:
+        fi
+      fi
+
+      for key in $(redis-cli --scan); do 
+        # Gets the first 8 digits fo the key to check if it belongs to Turbinia
+        key_type=$(echo "$key" | sed 's/:.*//')
+        if [[ "$key_type" == "TurbiniaTask" ]]; then
+          # Gets the Task value and split its key:value pairs into an array
+          value=$(redis-cli get "$key")
+          modified_value="${value//\"/}"
+          modified_value="${modified_value//\{/}"
+          modified_value="${modified_value//\}/}"
+          IFS=',' read -r -a array <<< "$modified_value"
+
+          if [[ "$field_name" == "all" ]]; then
+            if [[ "$action" == "values" ]]; then
+              echo -e "$value\n"
+            else
+              if [[ "$action" == keys ]]; then echo "$key"; fi
+              key_array+=( "$key" )
+            fi
+          else
+            for pair in "${array[@]}"; do
+              # Splits pair into field and value based on separating comma
+              current_field=$(echo "$pair" | sed 's/:.*//' | xargs)
+              current_value=$(echo "$pair" | sed 's/.*://' | xargs)
+              if [[ "$current_field"  == "$field_name" ]]; then
```

Comment:
Are there two nested `if` blocks to compare current_field and current_value because they won't fit on one line?  Maybe they'll fit after removing the other blocks from comments above?

Reply:
I nested these because there is a "break" inside the "field_name", so that the loop breaks whenever the "field_name" is matched, even if the "field_value" isn't matched, to improve runtime.